src/service.rs

```diff
@@ -474,4 +519,34 @@ impl Service {
             Err(io::Error::last_os_error().into())
         }
     }
+
+    /// Get the service config from the system.
+    pub fn query_config(&self) -> Result<ServiceInfo> {
+        let mut data: Vec<u8>;
+        let mut bytes_needed: u32 = 0;
+        unsafe {
+            winsvc::QueryServiceConfigW(
```

Comment:
Based on samples it’s safer to check the return value of the function and get the `io::Error::last_os_error()` and compare it against `ERROR_INSUFFICIENT_BUFFER` before going aheadand allocating a buffer and querying the config.

Sorry if it’s too paranoid, winapi is not very easy to work with.

That’s the sample I looked at today: https://docs.microsoft.com/en-us/windows/desktop/services/querying-a-service-s-configuration

Reply:
this is good feedback, i'll clean it up

Reply:
added checking around the raw error type. not sure i'm 100% happy with the flow.